sdk/core/core-tracing/review/core-tracing.api.md

```diff
@@ -90,6 +90,7 @@ export interface TracingContext {
 
 // @public
 export interface TracingSpan {
+    addEvent?(name: string, attributes?: Record<string, unknown>, startTime?: Date): void;
```

Comment:
Interesting that the spec uses `TimeStamp` but the opentelemetry-js API uses `startTime`.  Also should we align with https://open-telemetry.github.io/opentelemetry-js/classes/_opentelemetry_sdk_trace_base.Span.html#addEvent?

Reply:
My thinking is that, just like we (Azure SDK) need to maintain backward compatibility and support a large number of usecases, so do the OTel devs.

Some of the API surface area they provide is flexible but unnecessarily so. `attributesOrStartTime` smells like a backwards compatible evolution which we don't need. The various time formats they allow for `startTime` also isn't necessary - we can just use dates in our SDK.

It's kinda how our APIs are super-flexible to support multiple use-cases but SDK developers can enforce stricter interfaces since they have a more specific usecase. Does that seem reasonable?

Reply:
Ah I was thinking about the convenience of overloads.  How about an options bag with two optionals? It would be a bit awkward having to passing an `undefined` then a timestamp.

Reply:
oh yeah, that's a great call! Will do

Reply:
Updated to use `options` - thanks, can't believe I glossed over that!